### ISSUE TITLE
Accessibility: the labelText prop for DatePickerInput should be required

### ISSUE THREAD

**Issue by jendowns** [carbon-design-system/carbon-components-react#1635]
## Detailed description

The `labelText` prop for the DatePickerInput is currently not required: https://github.com/IBM/carbon-components-react/blob/master/src/components/DatePickerInput/DatePickerInput.js#L27

End users are therefore able to use `DatePicker` and the child `DatePickerInput` without providing any assistive text.

So you can have a situation like in the range variation of the `DatePicker` in Carbon React's storybook, where there are no labels provided to describe what the form inputs do: http://react.carbondesignsystem.com/?selectedKind=DatePicker&selectedStory=range%20with%20calendar%20and%20min%2Fmax%20dates

i.e. -- 
```html
<DatePicker
  id="date-picker"
  onChange={anonymous}
  minDate="1/10/2020"
  maxDate="1/20/2020"
  datePickerType="range"
>
   <DatePickerInput id="date-picker-input-id" light={false} onChange={anonymous} />
   <DatePickerInput id="date-picker-input-id-2" light={false} onChange={anonymous} />
</DatePicker>
```

For some reason, the `a11y` addon isn't flagging this component variation in Carbon React's storybook (perhaps due to the way the props object is passed in, or different storybook versions?), but the `a11y` addon *does* flag it in our environment:
```
#date-picker-input-id
    ✘ aria-label attribute does not exist or is empty
    ✘  aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty or not visible
    ✘ Form element does not have an implicit (wrapped) <label>
    ✘ Form element does not have an explicit <label>
    ✘ Element has no title attribute or the title attribute is empty

#date-picker-input-id-2
    ✘ aria-label attribute does not exist or is empty
    ✘ aria-labelledby attribute does not exist, references elements that do not exist or references elements that are empty or not visible
    ✘ Form element does not have an implicit (wrapped) <label>
    ✘ Form element does not have an explicit <label>
    ✘ Element has no title attribute or the title attribute is empty
```

Inputs need to have labels to go with them: https://www.w3.org/WAI/tutorials/forms/labels/
Even if the label text is implied, there needs to be some assistive text present (even if it's visually hidden).

By comparison, the `labelText` prop for the `TextInput` component is required: https://github.com/IBM/carbon-components-react/blob/master/src/components/TextInput/TextInput.js#L111

i.e. - 
```
  /**
   * Provide the text that will be read by a screen reader when visiting this
   * control
   */
labelText: PropTypes.node.isRequired,
```

Considering that form elements must have assistive labels (even if visually hidden), I propose requiring the `labelText` prop for the `DatePickerInput` component. If someone sets `hideLabel: true` (https://github.com/IBM/carbon-components-react/blob/master/src/components/TextInput/TextInput.js#L143), then perhaps the required `labelText` in the input should be moved to an `aria-label`

Thoughts?

**Comment by jendowns** [comment]
Just wanted to add: I'm happy to help with this. 

As I said toward the bottom of my description:
> I propose requiring the labelText prop for the DatePickerInput component. If someone sets hideLabel: true (https://github.com/IBM/carbon-components-react/blob/master/src/components/TextInput/TextInput.js#L143), then perhaps the required labelText in the input should be moved to an aria-label

But I'm open to other thoughts on this. Thank you! 🙏

**Comment by jendowns** [comment]
Note: the issue I referenced above, where I was seeing labels not included in one of the stories for the `DatePicker` is caused by this issue: https://github.com/IBM/carbon-components-react/issues/1637

**Comment by jendowns** [comment]
Hey @dakahn I was wondering what your thoughts are on this? (Note that the storybook has been updated to include `labelText` in the example, but that prop is still not required)

**Comment by dakahn** [comment]
i don't see any reason to have `labelText` be optional. 👍 @asudoh thoughts?

**Comment by asudoh** [comment]
@jendowns @dakahn Yeah if that prop is absolutely necessary to meet our a11y requirement let's make it required!

**Comment by dakahn** [comment]
not stale ⚡️

**Comment by dakahn** [comment]
@jendowns were you still able to help out on this 😸

**Comment by jendowns** [comment]
@dakahn Sure thing! PR incoming 😎
